modules/photo/include/opencv2/ccm.hpp

```diff
@@ -0,0 +1,520 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+//
+//                       License Agreement
+//              For Open Source Computer Vision Library
+//
+// Copyright(C) 2020, Huawei Technologies Co.,Ltd. All rights reserved.
+// Third party copyrights are property of their respective owners.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//             http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author: Longbu Wang <wanglongbu@huawei.com.com>
+//         Jinheng Zhang <zhangjinheng1@huawei.com>
+//         Chenqi Shan <shanchenqi@huawei.com>
+
+#ifndef OPENCV_PHOTO_CCM_HPP
+#define OPENCV_PHOTO_CCM_HPP
+
+#include <opencv2/core.hpp>
+#include <opencv2/imgproc.hpp>
+
+namespace cv
+{
+namespace ccm
+{
+
+/** @defgroup ccm Color Correction module
+@{
+
+Introduction
+------------
+
+The purpose of color correction is to adjust the color response of input
+and output devices to a known state. The device being calibrated is sometimes
+called the calibration source; the color space used as the standard is sometimes
+called the calibration target. Color calibration has been used in many industries,
+such as television production, games, photography, engineering, chemistry,
+medicine, etc. Due to the manufacturing process of the input and output equipment,
+the channel response has nonlinear distortion. In order to correct the picture output
+of the equipment, it is nessary to calibrate the captured color and the actual color.
+
+*/
+
+
+
+/** @brief Enum of the possible types of ccm.
+*/
+enum CCM_TYPE
+{
+    CCM_3x3,   ///< The CCM with the shape \f$3\times3\f$ performs linear transformation on color values.
+    CCM_4x3,   ///< The CCM with the shape \f$4\times3\f$ performs affine transformation.
```

Comment:
We need more worbose names and some description, what the difference between the cases.